syntax = "proto3";

package presentation;

// The service definition
service ProcessPresentationService {
  // Starts the processing chain
  rpc ProcessPresentation (ProcessPresentationRequest) returns (ProcessPresentationResponse);

  // Fetch the task status
  rpc GetTaskStatus (TaskStatusRequest) returns (TaskStatusResponse);
}

// The request message for starting a task
message ProcessPresentationRequest {
  string filepath = 1;
  string user_id = 2;
}

// The response message for starting a task
message ProcessPresentationResponse {
  string task_id = 1;
}

// The request message for fetching task status
message TaskStatusRequest {
  string task_id = 1;
}

// The response message for fetching task status
message TaskStatusResponse {
  string status = 1;
  string result = 2;
}
